### PR TITLE
PT-636 Feature/add config check route

### PR DESCRIPTION
**Pull request by BigFab**
I added new route for sanity check email sending

System tests were added. I leveraged what is done today in hydrophone for system tests. This could be done better, especially the shoreline mock that is not able to deal with wrong tokens (only present/absent).

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,6 +1,20 @@
 Hydrophone Dev Docs
 ===
 
+# Diabeloop added routes
+
+The below routes are added by Diabeloop to extend Hydrophone@tidepool service.
+
+## /inform
```
Comment (yvon-dblg) [inline docs/README.md]:
```suggestion
## POST /inform
```

File: docs/README.md [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
+
+This route is used to send an informative email to the recipient to let him know a patient account has been created for him. This is used for autoconfirmation of patient account creation from the DBL handset.
+
+## /sanity_check
```
Comment (yvon-dblg) [inline docs/README.md]:
```suggestion
## POST /sanity_check
```

File: api/verify.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+// Send a test email to prove all configuration is in place for sending emails
+//
+// status: 200
+// status: 500 internal server error
+func (a *Api) sendSanityCheckEmail(res http.ResponseWriter, req *http.Request, vars map[string]string) {
+
+	log.Printf("Sanity check email route")
+
+	var recipient string = a.Config.TestEmail
+	var subject string = "Sanity Check Email"
```
Comment (yvon-dblg) [inline api/verify.go]:
```suggestion
	subject := "Sanity Check Email"
```

File: api/verify.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+// Send a test email to prove all configuration is in place for sending emails
+//
+// status: 200
+// status: 500 internal server error
+func (a *Api) sendSanityCheckEmail(res http.ResponseWriter, req *http.Request, vars map[string]string) {
+
+	log.Printf("Sanity check email route")
+
+	var recipient string = a.Config.TestEmail
```
Comment (yvon-dblg) [inline api/verify.go]:
```suggestion
	recipient := a.Config.TestEmail
```

File: api/verify.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	var subject string = "Sanity Check Email"
+	var body string = "This is an automatic email sent from Hydrophone service to prove all configuration is in place for sending emails"
```
Comment (yvon-dblg) [inline api/verify.go]:
```suggestion
	body := "This is an automatic email sent from Hydrophone service to prove all configuration is in place for sending emails"
```

**Review by yvon-dblg — CHANGES_REQUESTED** [review]

**Comment by ydrisR** [comment]
@BigFab would be nice if the destination email was the email of the account corresponding to the token provided. i.e. we setup a test account (with our own email address) to execute this test during deployment. 
This would avoid spam, otherwise a user with a valid account could repeatidly call these route and send us thousands of email...

**Review by ydrisR — CHANGES_REQUESTED** [review]
@BigFab would be nice if the destination email was the email of the account corresponding to the token provided. i.e. we setup a test account (with our own email address) to execute this test during deployment.
This would avoid spam, otherwise a user with a valid account could repeatidly call these route and send us thousands of email...

**Comment by BigFab** [comment]
I reviewed the code to include your comments @yvon-dblg 
I also reviewed the code to make your suggestion a reality @ydrisR 
I finally reviewed the tests to accomodate new logic

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
- Need to be rebase from the dblp branch
- Error while testing in postman using the `portal.admin@diabeloop.fr` account
```
2019/09/24 12:26:22 Sanity check email route
2019/09/24 12:26:22 Found token in request header eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkdXIiOjI1OTIwMDAsImV4cCI6MTU3MTkxOTk3MSwic3ZyIjoibm8iLCJ1c3IiOiJlNmE5MjRiOTg4In0.GbyL0GQazv5II8_2GTJKDy3p7C5T8oL55pwxLPRb6jU
2019/09/24 12:26:23 Issue sending sanity check email: Status [500] Message [: 
	status code: 404, request id: ]
```

**Review by yvon-dblg — APPROVED** [review]

**Review by ydrisR — COMMENTED** [review]
Any reason to change the configuration file? (env.sh). It makes one more difference with tidepool origin...

**Comment by BigFab** [comment]
@ydrisR so about env.sh, do you mean we should not modify at all? This configuration file is not used by our stack, right?

**Comment by BigFab** [comment]
@ydrisR I did cancel the changes in the env.sh

**Review by ydrisR — APPROVED** [review]

**Pull request merged**